### PR TITLE
Use floats to represent intended fixed position

### LINKED ISSUE #1516: position_fixed reftest identical to its reference
Sorry I’m late to the party, but I notice that `src/test/ref/position_fixed_a.html` and `src/test/ref/position_fixed_b.html` (added in #1440) are identical.

This really doesn’t work, as a reftest. The idea is to use the feature being tested in the test page, and have a reference page that does *not* use the feature but should render identically. A bug in the feature being tested will (hopefully) make the test render differently from its reference.

@ibnc, please try to modify these files to make them a reftest. Sometimes coming up with a reference is hard, especially when we have so few layout primitives that are implemented yet. Maybe the reference here could use negative margins. (Do we support those?) If a reftest proves too difficult, please remove the files.

### PR DESCRIPTION
**Pull request by mitchhentges**
Thank you for contributing to Servo! Please replace each `[ ]` by `[X]` when the step is complete, and replace `__` with appropriate data:
- [X] `./mach build -d` does not report any errors
- [X] `./mach test-tidy --faster` does not report any errors
- [X] These changes fix #1516

Either:
- [X] There are tests for these changes

Pull requests that do not address these steps are welcome, but they will require additional verification as part of the review process.

---

This change is [Reviewable](https://reviewable.io/reviews/servo/servo/11330)

### CODE REVIEW

**Comment by KiChjang** [comment]
Fails the WPT lint:

```
diff --git a/tests/wpt/mozilla/meta/MANIFEST.json b/tests/wpt/mozilla/meta/MANIFEST.json

index 1b3f596..04adcb1 100644

--- a/tests/wpt/mozilla/meta/MANIFEST.json

+++ b/tests/wpt/mozilla/meta/MANIFEST.json

@@ -4108,16 +4108,16 @@

             "url": "/_mozilla/css/position_abs_width_percentage_a.html"

           }

         ],

-        "css/position_fixed_a.html": [

+        "css/position_fixed.html": [

           {

-            "path": "css/position_fixed_a.html",

+            "path": "css/position_fixed.html",

             "references": [

               [

-                "/_mozilla/css/position_fixed_b.html",

+                "/_mozilla/css/position_fixed_ref.html",

                 "=="

               ]

             ],

-            "url": "/_mozilla/css/position_fixed_a.html"

+            "url": "/_mozilla/css/position_fixed.html"

           }

         ],

         "css/position_fixed_background_color_a.html": [

@@ -10920,16 +10920,16 @@

           "url": "/_mozilla/css/position_abs_width_percentage_a.html"

         }

       ],

-      "css/position_fixed_a.html": [

+      "css/position_fixed.html": [

         {

-          "path": "css/position_fixed_a.html",

+          "path": "css/position_fixed.html",

           "references": [

             [

-              "/_mozilla/css/position_fixed_b.html",

+              "/_mozilla/css/position_fixed_ref.html",

               "=="

             ]

           ],

-          "url": "/_mozilla/css/position_fixed_a.html"

+          "url": "/_mozilla/css/position_fixed.html"

         }

       ],

       "css/position_fixed_background_color_a.html": [
```

File: tests/wpt/mozilla/tests/css/position_fixed_b.html [hunk 1]
```diff
@@ -7,37 +7,32 @@
...
-      right: 10px;
-      top: 100px;
-      bottom: 30px;
-    }
-    .stretched_fixed_block {
-      position: fixed;
-      background: black;
-      top: 100px;
-      bottom: 30px;
+      float:left;
```
Comment (KiChjang) [inline tests/wpt/mozilla/tests/css/position_fixed_b.html]: nit: Space after `:`.

**Comment by KiChjang** [comment]
r=me post-nit.

**Comment by mitchhentges** [comment]
Thanks for catching that, sorry for missing it

**Comment by KiChjang** [comment]
@bors-servo r+

Thanks!

**Comment by ConnorGBrewster** [comment]
@bors-servo retry
 * #11396

**Comment by ConnorGBrewster** [comment]
@bors-servo retry
 * #9106

**Comment by mbrubeck** [comment]
@bors-servo r- force

This PR re-enables `position_fixed_a.html` so we shouldn't check it in if it's failing.

**Comment by ConnorGBrewster** [comment]
@mbrubeck ahh good catch, sorry about that!

**Comment by mitchhentges** [comment]
That is ... really weird. I ran it ~50 times locally, and it passed every time. Maybe there's something special with the Mac build, or build agent? Can I get any more information on the failure?

**Comment by mitchhentges** [comment]
Found the issue with Mac: It is doing something funky with margins or floats (or maybe the combination). Either way, it works nicely on Windows and Linux, so I'm just making OSX an expected fail

position_fixed_a
position_fixed_b

**Comment by jdm** [comment]
Please file an issue about the mac failure so we don't forget about it :)

**Comment by mitchhentges** [comment]
Just changed the `ini` so check if `mac`, not if `osx`

**Comment by mitchhentges** [comment]
Hmm, I attempted to reproduce this locally, but I could no longer build `servo` on my branch. Once I merged master it, it built again, and the tests appear to pass.

I'm not really sure why they're failure on homu, maybe merging master in will help. At any rate, maybe merging master in will fix the Appveyor build failing

**Comment by mitchhentges** [comment]
So, this was failing on CI (and on Mac) because a different font was used, and that changes what the required offsets are.

I've updated this to use the same 'Ahem' font, because that's packaged with Servo. To attempt to circumvent #11482, I made each "block" only one letter. I now get the following two renders:

> position_fixed_a.html

2016-05-28-114651_800x600_scrot

> position_fixed_b.html

2016-05-28-114659_800x600_scrot


Yet, according to `./mach test-wpt tests/wpt/mozilla/tests/css/position_fixed_a.html`:
```
./mach test-wpt tests/wpt/mozilla/tests/css/position_fixed_a.html
Running 1 tests in web-platform-tests

  ▶ FAIL [expected PASS] /_mozilla/css/position_fixed_a.html
  └   → /_mozilla/css/position_fixed_a.html 0d34c4f90ff99d14010c7cc328367b84114fb0a8
/_mozilla/css/position_fixed_b.html 07ab080e2b7794ca3f893c59d8fcdd63de190af0
Testing 0d34c4f90ff99d14010c7cc328367b84114fb0a8 == 07ab080e2b7794ca3f893c59d8fcdd63de190af0

Unsupported test type wdspec for product servo
Ran 1 tests finished in 8.0 seconds.
  • 0 ran as expected. 0 tests skipped.
  • 1 tests failed unexpectedly
```

I can't find the difference between these images, and I'm not sure if there's something else going on.

**Comment by ConnorGBrewster** [comment]
@mitchhentges if you run `./mach test-wpt tests/wpt/mozilla/tests/css/position_fixed_a.html --log-raw=/pick/path/for/log.log` and then upload that log to the [reftest analyzer](http://hoppipolla.co.uk/410/reftest-analyser-structured.xhtml) you can see what the difference is between the two.

**Comment by mitchhentges** [comment]
Apparently the font isn't loaded at all in `test-wpt`?
no-font

File: log [hunk 2]
```diff
@@ -0,0 +1,95 @@
+{"source": "web-platform-tests", "thread": "MainThread", "time": 1464458627264, "action": "log", "message": "Using 6 client processes", "level": "INFO", "pid": 3625}
```
Comment (emilio) [inline log]: This file would need to go before landing this btw

**Comment by mitchhentges** [comment]
So, I've updated the tests to assume that the font will be rendered as black squres. Though two html files won't match up with `./mach run` (because 'Ahem' is actually rendered), it will be rendered consistently as black squares on all machines.

It's OK that the font is a black screen, because this test checks the positioning of fixed and dynamically-sized containers. Since even black-square-rendered text is still dynamically sized, the test is still valid

**Comment by emilio** [comment]
BTW, reading this now, I *think* Ahem is a square font, that's why it's useful for testing :-)

**Comment by mitchhentges** [comment]
That would explain a lot of things. Maybe `./mach run` is the thing that doesn't include the font?

I'll update the comment if we can confirm that Ahem is square-y. I thought it was a bug on my end that `gnome-font-viewer` was seeing squares

**Comment by emilio** [comment]
Whoops, sorry, deleted my previous comment thinking I had misread you but...

Yes, Ahem is AFAIK square-y, that's the point and why it's used for testing :-)

**Comment by emilio** [comment]
I think @mbrubeck can confirm it if you need more confirmation

**Comment by jdm** [comment]
Yes, Ahem doesn't contain actual letters, only solid squares.

**Comment by mitchhentges** [comment]
I think that it's totally good now

**Pull request merged**